Many SNEKs: schain, digitrust, pixel beacons, 3.0 compliance

aprakash-sovrn opened the pull request:
## Type of change
<!-- Remove items that don't apply and/or select an item by changing [ ] to [x] -->
- [ ] Bugfix
- [X] Feature
- [ ] New bidder adapter  <!--  IMPORTANT: if checking here, also submit your bidder params documentation here https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders --> 
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
Added schain, digitrust, and pixel sync'ing to the sovrn adapter
Upgraded adapter to be 3.0 compliant

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

aprakash-sovrn committed:
schain and digitrust

aprakash-sovrn committed:
pixel beacons

jrosendahl reviewed (commented):
tests?

TinchoF committed:
Outcon bid adapter. (#4161)

* Outcon bid adapter.

* Fix identation

* Fixes

* Fixes

* Fixes

* Spec fixes

* Fixes

* Fix urls

* Fix

* Fix parameters

* Fix space operators

* Fix bidder timeout

* Update

* Fix whitespace

* no message

* Outcon unit test

* no message

* no message

* no message

* no message

* Fixes

* Fixes

* Change url

* no message

* no message

* no message

* Added bidId

* no message

* no message

* no message

* no message

* Wrapping url with html

* no message

* no message

* no message

jaiminpanchal27 committed:
Adding workflow to run end to end tests (#4230)

* Adding workflow to run end to end tests

* trying self branch

* Update to run at 12 every day

* cleanup config using aliases

* update branch and cron time

* add command

aprakash-sovrn committed:
unit tests and fixes from testing

jaiminpanchal27 committed:
update prebid path for e2e test pages (#4274)

Commit:
Prebid 2.35.0 release

Commit:
Increment pre version

jrosendahl reviewed (approved):
looks good to me.

seergiioo6 committed:
Add usersync to adpone adapter (#4245)

* add user sync to adpone adapter

* move adpone usersync to global variable

* added withcredentials to http request

* fix http request options

* fix http request options

* add withCredentials: true

* add withCredentials: true

* added test coverage to usersync

* update sync function

* add test coverage

* adpone adapter

* package lock

* add more testing

* add more testing

* testing for onBidWon fucntion

* test onbidwon function

* trigger build

WayneYang1 committed:
Revert GumGum Adapter 2.28 resizing changes (#4277)

* changed resizing unit tests to return the first size dimensions in the sizes array

* added some changes

* reverted adapter changes

codybonney committed:
SpotX Bid Adapter: Support schain, ID5 object, Google consent object, and hide_skin (#4281)

* Add SpotXBidAdapter

* Minor updates

* Undo testing changes to shared files

* Fix relative imports

* Remove superfluous imports and write a few more tests

* Formatting, ID5 object, Google consent objects

  - Added ID5 object support
  - Added Google Consent object
  - Reformatted indentaiton on spec file

* Revert content_width and content_height changes in docs

  - not sure how these got moved, lets put them back

* Remove click_to_replay flag in example

  - no reason to use this one in the example

* Spotx adapter - Add schain support and update unit tests

* Update schain path in ORTB 2.3 request body

	- schain object is now added to ortb request body
	  at request.ext.source.ext.schain

* Add hide_skin to documentation

  - whoops, this got removed, let's add it back

Commit:
Update Rubicon Analytics Adapter `bidId` to match PBS (#4156)

* Add microadBidAdapter

* Remove unnecessary encodeURIComponent from microadBidAdapter

* Submit Advangelists Prebid Adapter

* Submit Advangelists Prebid Adapter 1.1

* Correct procudtion endpoint for prebid

* analytics update with wrapper name

* reverted error merge

* update for rubicon analytics to send seat[].bid.id for PBS video and banner

* fixed conditional for server and video or banner

* updated with optimized value test for bidid

* update changed default value of netRevenue to true

* remove var declaration for rightSlot to correct lgtm error for unused variable

* update defineSlot div id to match div id defined in html body

* update test ad unit test props

* revert lock to match remote master

* add seatBidId to bidObj in rpBidAdapter interpretResponse

* update setTargeting to execute in the bids back handler

* remove dev integration test page

* meaningless commit to get lgtm to re-run

evanmsmrtb committed:
SmartRTB adapter update (#4246)

* modules: Implement SmartRTB adapter and spec.

* Fix for-loop syntax to support IE; refactor getDomain out of exported set.

* Remove debugs, update doc

* Update test for video support

* Handle missing syncs. Add video to media types in sample ad unit

* Add null response check, update primary endpoint

* Note smrtb video requires renderer

tjeastmond committed:
Support Vast Track (#4276)

* Add microadBidAdapter

* Remove unnecessary encodeURIComponent from microadBidAdapter

* Submit Advangelists Prebid Adapter

* Submit Advangelists Prebid Adapter 1.1

* Correct procudtion endpoint for prebid

* analytics update with wrapper name

* reverted error merge

* update changed default value of netRevenue to true

* Add parameters if config.cache.vasttrack is true

* Use requestId instead of adId

* Test new vasttrack payload params

* Removed commented out code

* Relaxed conditional check per review

* Removed commented out line

hdeodhar committed:
Added 1000x250 size (#4295)

thewizarodofoz committed:
prepare vidazoo adapter for v3.0 (#4291)

jbartek25 committed:
Improve Digital adapter: support schain (#4286)

jankoulaga committed:
LiveIntent Identity Module. (#4178)

* LiveIntentIdSystem. Initial implementation.

* LiveIntentIdSystem. Removed whitespace.

* Fixed typo

* Renamed variables, cookiesm added md.

* Changed the default identity url.

* Composite id, with having more than just the lipbid passed around.

* Composite id.

* Merge conflict resolution.

* Changed docs and param description.

* Added typedoc & mentioned liveIntentIdSystem in submodule.json.

* Extracted the LiveIntentIdSystem under modules, removed it from default userId modules.

* Fixing the 204 + no body scenario.

* Added liveIntent to submodule.json

* Fixing docs indentation.

* Updated prebidServer & specs.

* Minor specs update.

bretg committed:
updating liveintent eids source (#4300)

* updating liveintent eids source

these are supposed to be domains

* updating unit test

weswhitney commented on modules/sovrnBidAdapter.js:
```diff
@@ -27,7 +27,22 @@ export const spec = {
+          }
+        }
+        if (bid.schain) {
+          schain = schain || bid.schain;
```
why `schain = schain || bid.schain;`? just curious

weswhitney reviewed (approved):
just a comment for learning discussion

jsnellbaker committed:
fix appnexusBidAdapter view-script regex (#4289)

* fix an view script regex

* minor syntax update

thomas-33across committed:
33Across adding bidder specific extension field (#4298)

* - add 33across specific ext field for statedAt

* - fix unit test for 33Across adapter

pm-harshad-mane committed:
PubMatic to support LiveIntent User Id sub-module (#4306)

* added support for pubcommon, digitrust, id5id

* added support for IdentityLink

* changed the source for id5

* added unit test cases

* changed source param for identityLink

* supporting LiveIntent Id in PubMatic adapter

* updated source for liveintent

finteza committed:
Finteza Analytics Adapter: fix cookies (#4292)

* fix reading and sending cookies

* fix lint errors

* clear comments

* add unit tests

* fix calling of setCookies for IE

* clear cookies after test

* use own setCookie method inside tests

margsliu committed:
Update LockerDome adapter to support Prebid 3.0 (#4301)

jankoulaga committed:
Returning the `IdResponse` type with an obj + callback. Fix for 4304 (#4305)

*  Returning the `IdResponse` type with an obj + callback.

* Renamed resp -> result.

* Removed whitespace.

vadim-mazzherin committed:
ShowHeroes adapter - expanded outstream support (#4222)

* add ShowHeroes Adapter

* ShowHeroes adapter - expanded outstream support

* Revert "ShowHeroes adapter - expanded outstream support"

This reverts commit bfcdb913b52012b5afbf95a84956b906518a4b51.

* ShowHeroes adapter - expanded outstream support

* ShowHeroes adapter - fixes (#4222)

* ShowHeroes adapter - banner and outstream fixes (#4222)

* ShowHeroes adapter - description and outstream changes (#4222)

* ShowHeroes adapter - increase test coverage and small fix

Commit:
[Orbidder-Adapter] Add bidRequestCount and remove bid.params.keyValues (#4264)

* initial orbidder version in personal github repo

* use adUnits from orbidder_example.html

* replace obsolete functions

* forgot to commit the test

* check if bidderRequest object is available

* try to fix weird safari/ie issue

* ebayK: add more params

* update orbidderBidAdapter.md

* use spec.<function> instead of this.<function> for consistency reasons

* add bidfloor parameter to params object

* fix gdpr object handling

* default to consentRequired: false when not explicitly given

* wip - use onSetTargeting callback

* add tests for onSetTargeting callback

* fix params and respective tests

* remove not used bid.params.keyValues

* add bidRequestCount to orbidder.otto.de/bid Post request

* add bidRequestCount to test object defaultBidRequest

anand-venkatraman committed:
PulsePoint: remove usage of deprecated utils method / prep for 3.0 (#4257)

* ET-1691: Pulsepoint Analytics adapter for Prebid. (#1)

* ET-1691: Adding pulsepoint analytics and tests for pulsepoint adapter

* ET-1691: Adding pulsepoint analytics and tests for pulsepoint adapter

* ET-1691: cleanup

* ET-1691: minor

* ET-1691: revert package.json change

* Adding bidRequest to bidFactory.createBid method as per https://github.com/prebid/Prebid.js/issues/509

* ET-1765: Adding support for additional params in PulsePoint adapter (#2)

* ET-1850: Fixing https://github.com/prebid/Prebid.js/issues/866

* Minor fix

* Adding mandatory parameters to Bid

* Removing usage of deprecated utils method

* minor refactor

jaiminpanchal27 committed:
Use isArray method (#4288)

eyas-ranjous committed:
Add Parrable ID submodule (#4266)

* add parrable id submodule

* fix integration test config

* fix var name

* always refresh sotredId for parrable

* add submodulesThatAlwaysRefresh concept

* remove comment

* add parrable url as one string

* add parrable prod endpoint

* use .indexOf instead of .includes

* add params to test config

* comment failing test

* uncomment failing assertion

* add parrable ID to prebid server adapter

* add parrableIdSystem to .submodules.json

* extract parrableId unit tests from userId spec

* remove breakline between imports

* remove unused param

* remove userId generic feature from parrableId module

* remove trailing space

* fix failing test due to none merged conflict

bretg committed:
Prebid 2.36.0 Release

bretg committed:
Increment pre version

madma committed:
Support schain module and send bidfloor param in Sharethrough adapter (#4271)

* Add support for supply chain object module

Story: [#168742394](https://www.pivotaltracker.com/story/show/168742394)

Co-authored-by: Josh Becker <jbecker@sharethrough.com>

* Add bidfloor parameter to bid request sent to STX

Story: [#168742573](https://www.pivotaltracker.com/story/show/168742573)

hbanalytics committed:
Platform One Analytics Adapter (#4233)

* Added Y1 Analytics Adapter

* rename y1AnalyticsAdapter in yieldoneAnalyticsAdapter

* Yieldone Bid Adapter: fixes from lint check

* Yieldone Analytics Adapter: fix endpoint protocol

* Added spec file for yieldone Analytics Adapter

eyas-ranjous committed:
Fix parrable id integration example (#4317)

* fix parrableId integration example

* add parentheses

jbartek25 committed:
Improve Digital adapter: support for video (#4318)

* Bid floor, https, native ad update

* Update the ad server protocol module

* Adding referrer

* Improve Digital support for video

* Improve Digital adapter: video

* adapter version -> 6.0.0

Commit:
Gamoshi: Update aliases list. Add support for userSync. (#4319)

* Add support for multi-format ad units. Add favoredMediaType property to params.

* Add tests for gdpr consent.

* Add adId to outbids

* Modify media type resolving

* Refactor multi-format ad units handler.

* Modify the way of sending GDPR data.
Update aliases.

* Add new consent fields. Add unit test.

* Add new consent fields. Add unit test.

* Add support for id5 and unified id cookie sync.

* Add support for id5 and unified id cookie sync.

* Add restricted check for gdpr consent.

aprakash-sovrn committed:
Prebid 3.0 updates

aprakash-sovrn committed:
Merge branch 'master' of https://github.com/prebid/Prebid.js
